web3/providers/persistent/persistent.py

```diff
@@ -50,6 +49,18 @@ def __init__(
         self.request_timeout = request_timeout
         self.silence_listener_task_exceptions = silence_listener_task_exceptions
 
+    @property
+    def endpoint_uri_or_ipc_path(self) -> str:
+        if hasattr(self, "endpoint_uri"):
+            return str(self.endpoint_uri)
+        elif hasattr(self, "ipc_path"):
+            return str(self.ipc_path)
+        else:
+            raise AttributeError(
```

Comment:
depending on when this gets merged, may need to swap it for a `Web3AttributeError`